test/legend-visualMapColor.html

```diff
@@ -0,0 +1,209 @@
+                "legend": {
+                    "show": true
+                }
+            };
+
+            var chart = testHelper.create(echarts, 'main0', {
+                title: [
+                    'Legend use color from visualMap',
+                    'Test case from #10766'
+                ],
+                option: option
+            });
+        });
+        </script>
+
+
+        <script>
+        require(['echarts'/*, 'map/js/china' */], function (echarts) {
+            var data = [
+                {value:335, name:'直接访问'},
+                {value:310, name:'邮件营销'},
+                {value:274, name:'联盟广告'},
+                {value:235, name:'视频广告'},
+                {value:400, name:'搜索引擎'}
+            ].sort(function (a, b) { return a.value - b.value; });
+            var option = {
+                backgroundColor: '#2c343c',
+
+                legend: {
+                    data: data.map(function (item) {
+                        return item.name
+                    }),
+                    textStyle: {
+                        color: '#fff'
+                    }
+                },
+                visualMap: {
+                    show: false,
+                    min: 80,
+                    max: 600,
+                    inRange: {
+                        colorLightness: [0, 1]
+                    }
+                },
+                series : [
+                    {
+                        name:'访问来源',
+                        type:'pie',
+                        radius : '55%',
+                        center: ['50%', '50%'],
+                        data: data,
+                        roseType: 'radius',
+                        label: {
+                            normal: {
+                                textStyle: {
+                                    color: 'rgba(255, 255, 255, 0.3)'
+                                }
+                            }
+                        },
+                        labelLine: {
+                            normal: {
+                                lineStyle: {
+                                    color: 'rgba(255, 255, 255, 0.3)'
+                                },
+                                smooth: 0.2,
+                                length: 10,
+                                length2: 20
+                            }
+                        },
+                        itemStyle: {
+                            normal: {
+                                color: '#c23531',
+                                shadowBlur: 200,
+                                shadowColor: 'rgba(0, 0, 0, 0.5)'
+                            }
+                        },
+
+                        animationType: 'scale',
+                        animationEasing: 'elasticOut',
+                        animationDelay: function (idx) {
+                            return Math.random() * 200;
+                        }
+                    }
+                ]
+            };
```

Comment:
Do we need to add (or already have?) a test case like #2372 described (I know it is correct both before and after this change).
Like:
```js
option = {
    legend: {
    },
    series: {
        type: 'pie',
        itemStyle: {
            color: function (params) {
                var colorList = ['#ff3322', '#232211', '#aabbcc'];
                return colorList[params.dataIndex];
            }
        },
        data: [
            {name: 'a', value: 1222},
            {name: 'b', value: 2333},
            {name: 'c', value: 3444},
        ]
    }
};
```